oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/Evaluation.java

```diff
@@ -0,0 +1,154 @@
+
+            double dbIndex = (clusterScatter1 + clusterScatter2) / getDistance(c1.getCenter(), c2.getCenter());
+
+            if (dbIndex > maxDBIndex) {
+              maxDBIndex = dbIndex;
+            }
+          }
+        }
+      }
+      totalDBIndex += maxDBIndex;
+    }
+
+    return totalDBIndex / clusters.size();
+  }
+
+  private static double calcDunnIndex(List<Tuple2<Integer,Tuple2<Double,Long>>> clusterSumDistAndCounts,
+                                      List<ClusterInfo> clusters) {
+    double maxIntraClusterDistance = 0.0;
+    for (Tuple2<Integer,Tuple2<Double,Long>> entry : clusterSumDistAndCounts) {
+      double intraClusterDistance = entry._2()._1() / entry._2()._2();
+      if (maxIntraClusterDistance < intraClusterDistance) {
+        maxIntraClusterDistance = intraClusterDistance;
+      }
+    }
+
+    double minInterClusterDistance = Double.POSITIVE_INFINITY;
+    for (int i = 0; i < clusters.size(); i++) {
+      // Distances are symmetric, hence d(i,j) == d(j,i)
+      for (ClusterInfo cluster : clusters) {
+        double distance = getDistance(clusters.get(i).getCenter(), cluster.getCenter());
+        if (minInterClusterDistance > distance) {
+          minInterClusterDistance = distance;
+        }
+      }
+    }
+
+    return minInterClusterDistance / maxIntraClusterDistance;
+
+  }
+
+  private static JavaPairRDD<Integer, Tuple2<Double,Long>> fetchClusterSumDistanceAndCounts(
+      final List<ClusterInfo> clusters, JavaRDD<Vector> testData) {
+
+    return testData.mapToPair(new PairFunction<Vector, Integer, Tuple2<Double, Long>>() {
+      @Override
+      public Tuple2<Integer, Tuple2<Double, Long>> call(Vector vector) {
+        double closestDist = Double.POSITIVE_INFINITY;
+        int minCluster = -1;
+        for (int i = 0; i < clusters.size(); i++) {
+          ClusterInfo cluster = clusters.get(i);
+          double distance = getDistance(cluster.getCenter(), vector.toArray());
+          if (distance < closestDist) {
+            closestDist = distance;
+            minCluster = i;
+          }
+        }
+        Preconditions.checkState(minCluster >= 0);
+        Preconditions.checkState(!Double.isInfinite(closestDist) && !Double.isNaN(closestDist));
+        return new Tuple2<>(minCluster, new Tuple2<>(closestDist, 1L));
+      }
+    }).reduceByKey(
+        new Function2<Tuple2<Double, Long>, Tuple2<Double, Long>, Tuple2<Double, Long>>() {
+          @Override
+          public Tuple2<Double, Long> call(Tuple2<Double, Long> v1, Tuple2<Double, Long> v2) {
+            double intraClusterDistanceSum = v1._1() + v2._1();
+            long clusteredPointsCount = v1._2() + v2._2();
+
+            return new Tuple2<>(intraClusterDistanceSum, clusteredPointsCount);
+          }
+        });
+  }
+
+  private static double getDistance(double[] v1, double[] v2) {
+    SquaredDistanceFn distanceFn = new SquaredDistanceFn();
```

Comment:
I feel like this should be a member and this can be a non-static function rather than allocate an object every time.